src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -0,0 +1,214 @@
+            IReviewManager reviewManager, IAPIRevisionsManager apiRevisionManager, ICommentsManager commentsManager,
+            ILogger<AutoReviewController> logger)
+        {
+            _authorizationService = authorizationService;
+            _codeFileManager = codeFileManager;
+            _apiRevisionsManager = apiRevisionManager;
+            _commentsManager = commentsManager;
+            _reviewManager = reviewManager;
+            _logger = logger;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> UploadAutoReview([FromForm] IFormFile file, string label, bool compareAllRevisions = false)
+        {
+            if (file != null)
+            {
+                using (var openReadStream = file.OpenReadStream())
+                {
+                    using var memoryStream = new MemoryStream();
+                    var codeFile = await _codeFileManager.CreateCodeFileAsync(originalName: file.FileName, fileStream: openReadStream,
+                        runAnalysis: false, memoryStream: memoryStream);
+
+                    var apiRevision = await CreateAutomaticRevisionAsync(codeFile: codeFile, label: label, originalName: file.FileName, memoryStream: memoryStream, compareAllRevisions);
+
+                    if (apiRevision != null)
+                    {
+                        var reviewUrl = $"{this.Request.Scheme}://{this.Request.Host}/Assemblies/Review/{apiRevision.ReviewId}?revisionId={apiRevision.Id}";
+                        return apiRevision.IsApproved ? Ok(reviewUrl) : StatusCode(statusCode: StatusCodes.Status201Created, reviewUrl);
+                    }
+                };
+            }
+            // Return internal server error for any unknown error
+            return StatusCode(statusCode: StatusCodes.Status500InternalServerError);
+        }
+    
+        public async Task<ActionResult> GetReviewStatus(string language, string packageName, string reviewId = null, bool? firstReleaseStatusOnly = null)
+        {
+            // This API is used by prepare release script to check if API review for a package is approved or not.
+            // This caller script doesn't have artifact to submit and so it can't check using create review API
+            // So it rely on approval status of latest revision of automatic review for the package
+            // With new restriction of creating automatic review only from master branch or GA version, this should ensure latest revision
+            // is infact the version intended to be released.
+
+            ReviewListItemModel review;
+            if (String.IsNullOrEmpty(reviewId))
+            {
+                review = await _reviewManager.GetReviewAsync(packageName: packageName, language: language, isClosed: null);
+            }
+            else
+            {
+                review = await _reviewManager.GetReviewAsync(User, reviewId);
```

Comment:
I think we don't need this anymore. Package name and Language are required parameters and fetching using those two params will lead to same exact review fetched using review ID.